packages/bun-uws/src/HttpParser.h

```diff
@@ -235,42 +241,25 @@ namespace uWS
         {
             return (x == '-') |
                    ((x > '/') & (x < ':')) |
-                   ((x > '@') & (x < '[')) |
-                   ((x > 96) & (x < '{'));
+                   isFieldNameFirstByte(x) |
+                   (x == '_') |
+                   (x == '.');
         }
 
         static inline uint64_t hasLess(uint64_t x, uint64_t n)
         {
             return (((x) - ~0ULL / 255 * (n)) & ~(x) & ~0ULL / 255 * 128);
         }
 
-        static inline uint64_t hasMore(uint64_t x, uint64_t n)
-        {
-            return ((((x) + ~0ULL / 255 * (127 - (n))) | (x)) & ~0ULL / 255 * 128);
-        }
-
-        static inline uint64_t hasBetween(uint64_t x, uint64_t m, uint64_t n)
-        {
-            return (((~0ULL / 255 * (127 + (n)) - ((x) & ~0ULL / 255 * 127)) & ~(x) & (((x) & ~0ULL / 255 * 127) + ~0ULL / 255 * (127 - (m)))) & ~0ULL / 255 * 128);
-        }
-
-        static inline bool notFieldNameWord(uint64_t x)
-        {
-            return hasLess(x, '-') |
-                   hasBetween(x, '-', '0') |
-                   hasBetween(x, '9', 'A') |
-                   hasBetween(x, 'Z', 'a') |
-                   hasMore(x, 'z');
-        }
-
         static inline void *consumeFieldName(char *p)
         {
             for (; true; p += 8)
             {
                 uint64_t word;
                 memcpy(&word, p, sizeof(uint64_t));
-                if (notFieldNameWord(word))
+                if (isFieldNameFirstByte(*(unsigned char *)p))
```

Comment:
This is too many changes. The only change here should be to check for `_` and `.`

Reply:
Yes, but per request @paperdave [link](https://discord.com/channels/876711213126520882/888839314056839309/1199998725997723748) i also added first character check for letter only

If it's extra, I can return it

Reply:
i was under the assumption the parser was already this way but it wasnt, i change my mind on this. sorry

Reply:
lets do just the added `|` to keep this as minimal as possible to reduce breaking something

Reply:
will it be enough if i change to

```js
static inline bool notFieldNameWord(uint64_t x)
        {
            return hasLess(x, '9') |
                   hasBetween(x, '9', 'A') |
                   hasBetween(x, 'Z', 'a') |
                   hasMore(x, 'z');
        }
```

Reply:
This will be a small enough change to not break anything, but enough to allow only letters

also a couple of cpu instructions less)
however, in my unprofessional opinion, my version has even fewer cpu instructions

Reply:
@Jarred-Sumner @paperdave

Reply:
Done it